yt/yt/tests/integration/queries/test_mock.py

```diff
@@ -560,3 +560,56 @@ def expect_queries(queries, list_result, incomplete=False):
         expect_queries([q_u3, q_u4], list_queries(cursor_direction="future", authenticated_user="u4"))
         expect_queries([q_u3, q_u5], list_queries(cursor_direction="future", authenticated_user="u5"))
         expect_queries([q_u6], list_queries(cursor_direction="future", authenticated_user="u6"))
+
+    @authors("krock21")
+    def test_list_by_aco(self, query_tracker):
+        create_user("u1")
+        create_user("u2")
+        create_user("u3")
+        create_access_control_object(
+            "aco_list_by_aco1",
+            "queries",
+            attributes={
+                "principal_acl": [
+                    make_ace("allow", "users", "use"),
+                ]
+            })
+        create_access_control_object(
+            "aco_list_by_aco2",
+            "queries",
+            attributes={
+                "principal_acl": [
+                    make_ace("allow", "users", "use"),
+                ]
+            })
+        create_access_control_object(
+            "aco_list_by_aco3",
+            "queries",
+            attributes={
+                "principal_acl": [
+                    make_ace("allow", "users", "use"),
+                ]
+            })
+        q_u1 = start_query("mock", "run_forever", authenticated_user="u1", access_control_object="aco_list_by_aco1")
+        q_u2 = start_query("mock", "run_forever", authenticated_user="u2", access_control_object="aco_list_by_aco2")
+        q_u3 = start_query("mock", "run_forever", authenticated_user="u3", access_control_object="aco_list_by_aco3")
+
+        def expect_queries(queries, list_result, incomplete=False):
+            try:
```

Comment:
We already have this function
https://github.com/ytsaurus/ytsaurus/blob/main/yt/yt/tests/integration/queries/test_mock.py#L113
https://github.com/ytsaurus/ytsaurus/blob/main/yt/yt/tests/integration/queries/test_mock.py#L528
Why do we need to copypaste it again? Maybe it can be generalized in some way?

Reply:
Extracted it out of test classes